coveo-functools/coveo_functools/flex/deserializer.py

```diff
@@ -24,24 +20,40 @@
 from coveo_functools.casing import TRANSLATION_TABLE, unflex
 from coveo_functools.dispatch import dispatch
 from coveo_functools.exceptions import UnsupportedAnnotation
+from coveo_functools.flex.helpers import resolve_hint
+from coveo_functools.flex.serializer import SerializationMetadata
 from coveo_functools.flex.subclass_adapter import get_subclass_adapter
-from coveo_functools.flex.types import TypeHint, PASSTHROUGH_TYPES
+from coveo_functools.flex.types import TypeHint, is_passthrough_type, PASSTHROUGH_TYPES
 
 
 T = TypeVar("T")
 
+MetaHint = Union[Callable[..., T], SerializationMetadata, Type[T]]
 
-def prepare_payload_for_unpacking(
-    fn: Callable[..., T], dirty_kwargs: Dict[str, Any]
-) -> Dict[str, Any]:
-    """Return a copy of `dirty_kwargs` that can be `**unpacked` to fn()."""
+
+def convert_kwargs_for_unpacking(dirty_kwargs: Dict[str, Any], *, hint: MetaHint) -> Dict[str, Any]:

```

Comment:
this is `prepare_payload_for_unpacking` renamed, and adjusted to work with the new metadata.